File: IRemember/app/src/main/java/ru/goodibunakov/iremember/alarm/AlarmSetter.java
package ru.goodibunakov.iremember.alarm;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import java.util.ArrayList;
import java.util.List;

import ru.goodibunakov.iremember.database.DbHelper;
import ru.goodibunakov.iremember.model.ModelTask;

public class AlarmSetter extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        DbHelper dbHelper = new DbHelper(context);

        AlarmHelper.getInstance().init();
        AlarmHelper alarmHelper = AlarmHelper.getInstance();
        List<ModelTask> tasks = new ArrayList<>(dbHelper.query().getTasks(DbHelper.SELECTION_STATUS + " OR "
                        + DbHelper.SELECTION_STATUS,
                new String[]{Integer.toString(ModelTask.STATUS_CURRENT), Integer.toString(ModelTask.STATUS_OVERDUE)},
                DbHelper.TASK_DATE_COLUMN));

        for (ModelTask task: tasks){
            if (task.getDate() != 0){
                alarmHelper.setAlarm(task);
            }
        }
    }
}

File: IRemember/app/src/main/java/ru/goodibunakov/iremember/alarm/AlarmSetter.kt
package ru.goodibunakov.iremember.alarm

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import ru.goodibunakov.iremember.RememberApp

class AlarmSetter : BroadcastReceiver() {

    lateinit var disposable: Disposable

    override fun onReceive(context: Context?, intent: Intent?) {
        if (intent?.action != "android.intent.action.BOOT_COMPLETED") return

        val alarmHelper = AlarmHelper.getInstance()
        alarmHelper.initAlarmManager()
        disposable = RememberApp.getBus().getEvent()
                .subscribe { it ->
                    RememberApp.databaseRepository.findCurrentTasks(it)
                            .subscribeOn(Schedulers.io())
                            .observeOn(AndroidSchedulers.mainThread())
                            .subscribe({
                                for (element in it) {
                                    if (element.date != 0L) {
                                        alarmHelper.setAlarm(element)
                                    }
                                }
                                dispose()
                            }, { error ->
                                Log.d("debug", "Ошибка в AlarmSetter ${error!!.localizedMessage!!}")
                            })
                }

//        RememberApp.getBus().post("")

//        val tasks = ArrayList(dbHelper.query().getTasks(
//                DbHelper.SELECTION_STATUS + " OR " + DbHelper.SELECTION_STATUS,
//                arrayOf(ModelTask.STATUS_CURRENT.toString(), ModelTask.STATUS_OVERDUE.toString()),
//                DbHelper.TASK_DATE_COLUMN)
//        )
//
//        for (task in tasks) {
//            if (task.date != 0L) {
//                alarmHelper.setAlarm(task)
//            }
//        }
    }

    private fun dispose() {
        if (::disposable.isInitialized && !disposable.isDisposed) disposable.dispose()
    }
}